samcli/lib/providers/sam_function_provider.py

```diff
@@ -390,10 +402,13 @@ def _parse_layer_info(
                         layer_logical_id, layer_properties, "ContentUri", ignore_code_extraction_warnings
                     )
 
+                if codeuri and not use_raw_codeuri:
+                    codeuri = SamLocalStackProvider.normalize_resource_path(stack.location, codeuri)
```

Comment:
Adding some debug statements might help to investigate further issues just in case.